### DISCUSSION TITLE
Gradle Build Plugin

### DISCUSSION THREAD

**Discussion by jason-visotrust** [open-telemetry/opentelemetry-java#3756] [Q&A]
I've built a Gradle plugin that exports spans for the gradle build, each task, each test suite, and each test. It uses the Gradle BuildListener, TaskExecutionListener, and TestListener interfaces to get callbacks. It uses your library with the autoconfiguration to set up exporters, etc. 

I was wondering if there would be interest in adding this to opentelemetry-java so the wider community could use it to track build times?

Thanks,
Jason

**Comment by jkwatson** [comment]
Something like this? https://github.com/jkwatson/javaagent-gradle-plugin  ;)

There is a maven build plugin like this in the `opentelemetry-java-contrib` repo, which is where I think this kind of thing belongs.

**Comment by jason-visotrust** [comment]
Nope! I just use the java agent at runtime for that. This is for instrumenting your build so you can see how long the different tasks in your gradle build take and how long the individual tests take. I have it exporting to Honeycomb so we can track how long our builds take and target the slowest running tests to improve build time.

**Reply by jkwatson** [reply]
Whoops! Linked to the wrong repo. How about this one : https://github.com/jkwatson/gradle-otel-tracing

**Reply by jason-visotrust** [reply]
Yep, like that. The big value for us is in the test timings, though. You should look at the Gradle TestListener interface.

**Reply by jkwatson** [reply]
If yours is better, then contributing it at the `opentelemetry-java-contrib` repository would be awesome. Mine was just a hackathon project. :)

**Comment by sureshg** [comment]
@jason-visotrust Is this available for public use now?
